apps/src/templates/rubrics/LearningGoal.jsx

```diff
@@ -38,6 +57,78 @@ export default function LearningGoal({
     setIsOpen(!isOpen);
   };
 
+  const handleFeedbackChange = event => {
+    if (studentLevelInfo.user_id && learningGoal.id) {
+      if (autosaveTimer.current) {
+        clearTimeout(autosaveTimer.current);
+      }
+      teacherFeedback.current = event.target.value;
+      setDisplayFeedback(teacherFeedback.current);
+      autosaveTimer.current = setTimeout(() => {
+        autosave();
+      }, saveAfter);
+    }
+  };
+
+  const autosave = () => {
+    setAutosaved(false);
+    setIsAutosaving(true);
+    setErrorAutosaving(false);
+    const bodyData = JSON.stringify({
+      studentId: studentLevelInfo.user_id,
+      learningGoalId: learningGoal.id,
+      feedback: teacherFeedback.current,
+      understanding: understandingLevel.current,
+    });
+    HttpClient.put(`${base_endpoint}/${learningGoalEval.id}`, bodyData, true, {
+      'Content-Type': 'application/json',
+    })
+      .then(() => {
+        setIsAutosaving(false);
+        setAutosaved(true);
+      })
+      .catch(error => {
+        console.log(error);
+        setIsAutosaving(false);
+        setErrorAutosaving(true);
+      });
+    clearTimeout(autosaveTimer.current);
+  };
+
+  useEffect(() => {
+    if (studentLevelInfo && learningGoal.id) {
+      const body = JSON.stringify({
+        userId: studentLevelInfo.user_id,
+        learningGoalId: learningGoal.id,
+      });
+      HttpClient.post(`${base_endpoint}/get_or_create_evaluation`, body, true, {
+        'Content-Type': 'application/json',
+      })
+        .then(response => response.json())
+        .then(json => {
+          setLearningGoalEval(json);
+          if (!json.feedback) {
+            teacherFeedback.current = '';
+          } else {
+            teacherFeedback.current = json.feedback;
+            setDisplayFeedback(teacherFeedback.current);
+          }
+          if (json.understanding) {
+            understandingLevel.current = json.understanding;
+          }
+        })
+        .catch(error => console.log(error));
+    }
+  }, [studentLevelInfo, learningGoal]);
+
+  // Callback to retrieve understanding data from EvidenceLevels
+  const radioButtonCallback = radioButtonData => {
+    understandingLevel.current = radioButtonData;
+    if (!isAutosaving) {
+      autosave();
```

Comment:
Is there a race condition here if an autosave in in progress and no other changes are made?